src/parser/utils.rs

```diff
@@ -583,6 +583,21 @@ pub fn mk_let(
     Ok(result)
 }
 
+/// Generate a `Fun` or a `FunPattern` (depending on `pat` being empty or not) from the
+/// parsing of a function definition. This function panics if the definition somehow
+/// has neither an `Ident` nor a non-`Empty` `Destruct` pattern.
+pub fn mk_fun(id: Option<Ident>, pat: Option<DestructuredRecord>, body: RichTerm) -> Term {
+    match pat {
+        Some(d) => Term::FunPattern(id, d, body),
+        None => {
+            let Some(id) = id else {
+                unreachable!("functions always have either a non-Empty pattern or an ident")
```

Comment:
Could we make an enum type to encode this information and avoid the `unreachable!` ? (and the `panic!` in `mk_let`). Something like

```
enum FunOrLetLeftHandSideButObviouslyPleaseFindAShorterName {
  Simple(id),
  Pattern(DestructureRecord),
}
```

Reply:
Yeah, I had this thought myself but I was being lazy & figured I could come back to it later. You're probably right that it makes sense to do it now though.